massa-ledger/src/ledger_db.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+use massa_hash::Hash;
+use massa_models::{Address, Amount, DeserializeVarInt, SerializeVarInt};
+use rocksdb::{
+    ColumnFamilyDescriptor, Direction, IteratorMode, Options, ReadOptions, WriteBatch, DB,
+};
+use std::{collections::BTreeMap, str::FromStr};
+
+use crate::{ledger_changes::LedgerEntryUpdate, LedgerEntry, SetOrDelete, SetOrKeep};
+
+// TODO: remove rocks_db dir when sled is cut out
+const DB_PATH: &str = "../massa-node/storage/ledger/rocks_db";
+const LEDGER_CF: &str = "ledger";
+const METADATA_CF: &str = "metadata";
+const OPEN_ERROR: &str = "critical: rocksdb open operation failed";
+const CRUD_ERROR: &str = "critical: rocksdb crud operation failed";
+const CF_ERROR: &str = "critical: rocksdb column family operation failed";
+
+/// Ledger sub entry enum
+pub enum LedgerSubEntry {
+    /// Balance
+    Balance,
+    /// Bytecode
+    Bytecode,
```

Comment:
in a future PR, we could fuse Bytecode into the datastore

Reply:
Yes I believe that should be part of the follow up as well.